Pull Request 1 (proj sps + proj beat that)

SJKKSJ opened the pull request:
random generator created & tested;
input validation unsure;
conditional stmts unsure

SJKKSJ committed:
initial commit

SJKKSJ committed:
Pre Class Day 3

SJKKSJ committed:
day 2 pre class revise

SJKKSJ committed:
Day 3 In Class
Base Secret Word & Comfortable Secret Word Twice in a Row

SJKKSJ committed:
SPS Part 1 Base (complete)

SJKKSJ committed:
SPS Part 2 (win loss)
added in the counter for wins/losses/draws/total games played +
win loss percentage

SJKKSJ committed:
SPS Part 1 (base) + Part 2 (win loss record)
updated the whole block of codes

SJKKSJ committed:
sps index html update 
project SPS index html update

SJKKSJ committed:
SPS Part 2 (Username)
Part 2 - Username

SJKKSJ committed:
SPS Part 2 (username + formatting)
> update the username part
> formatting break line to output message

SJKKSJ commented:
- How many hours did you spend on this assignment?: _+/-14hrs_
- What part of the assignment did you spend the most time on?: _main game logic and the game modes for username_
- What is one error and/or error message you received while working on this assignment, and how did you solve the error? _undefined and NaN errors, checked the variables names and called the right ones + return output_
- How comfortable did you feel with this assignment? (1-5): _3_
- Is there anything in this code that you feel pleased about?: _made the game logic conditions clearer and that the game works_
- What's one aspect of your code you would like specific, elaborate feedback on?: _game logic conditions / the return percentage output_

lee-kimmixq commented on project1-sps/index.html:
```diff
@@ -72,7 +72,7 @@ <h1 id="header">🚀 Basics Project 1: Scissors Paper Stone 🚀</h1>
       <button id="submit-button">Submit</button>
       <p>Output:</p>
-      <div class="output-div" id="output-div"></div>
+      <div class="output-div" id="output-div">Hi! Please enter your name.</div>
```
Nice!

lee-kimmixq commented on project1-sps/script.js:
```diff
@@ -1,4 +1,92 @@
+var userDrawRecord = 0;
+
+// computerOption helper function
+var computerSPS = function (input) {
```
Nice helper function! Maybe you can name it starting with a verb, something like `getComputerSPS`

lee-kimmixq commented on project1-sps/script.js:
```diff
@@ -1,4 +1,92 @@
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue;
```
Since we are not using `myOutputValue` in the first game mode, we can define it inside the code block of the second game mode (so that we don't define the variable _"for nothing"_)

lee-kimmixq commented on project1-sps/script.js:
```diff
@@ -1,4 +1,92 @@
+  } else if (currentMode === "Start Game") {
+    // user input validation
+    // if user input not equals to scissors/paper/stone, run invalid msg
+    if (!input == "scissors" || "paper" || "stone") {
+      var myOutputValue =
+        "Please enter a valid game input - scissors, paper, stone";
+    }
```
Nice input validation here! Since we are defining `myOutputValue` before this, line 35 should be a reassignment - no need to repeat the keyword var

lee-kimmixq commented on project1-sps/script.js:
```diff
@@ -1,4 +1,92 @@
+    var computerOption = computerSPS();
+    // game win or lose conditions
+    // if player and computer same option, draw
+    if (playerOption == computerOption) {
+      myOutputValue = "It's a draw.";
+      userDrawRecord = userDrawRecord + 1;
+    }
+    // Player puts out scissors, if computer plays paper, user wins, else lose
+    else if (playerOption == "scissors") {
+      if (computerOption == "paper") {
+        myOutputValue = "You win! ";
+        userWinRecord = userWinRecord + 1;
+      } else {
+        myOutputValue = "You lose! ";
+        userLoseRecord = userLoseRecord + 1;
+      }
+    }
+    // Player puts out paper, if computer plays stone, user wins, else lose
+    else if (playerOption == "paper") {
+      if (computerOption == "stone") {
+        myOutputValue = "You win! ";
+        userWinRecord = userWinRecord + 1;
+      } else {
+        myOutputValue = "You lose! ";
+        userLoseRecord = userLoseRecord + 1;
+      }
+    }
+    // Player puts out stone, if computer plays scissors, user wins, else lose
+    else if (playerOption == "stone") {
+      if (computerOption == "scissors") {
+        myOutputValue = "You win! ";
+        userWinRecord = userWinRecord + 1;
+      } else {
+        myOutputValue = "You lose! ";
+        userLoseRecord = userLoseRecord + 1;
+      }
+    }
```
Good use of conditionals here! You might want to use the && and || operators to simplify the current nested if-else statements :) Also will be good to refactor this into a helper function!

lee-kimmixq commented on project1-sps/script.js:
```diff
@@ -1,4 +1,92 @@
+      }
+    }
+    // total number of games played, to calculate win loss percentage
+    var totalGamesPlayed = userWinRecord + userLoseRecord + userDrawRecord;
+
+    return `Hi ${userName}! <br> ${myOutputValue} <br> Wins: ${userWinRecord}(${(
+      (userWinRecord * 100) /
+      totalGamesPlayed
+    ).toFixed(2)}%) <br> Losses: ${userLoseRecord}(${(
+      (userLoseRecord * 100) /
+      totalGamesPlayed
+    ).toFixed(2)}%) <br> Draws: ${userDrawRecord}(${(
+      (userDrawRecord * 100) /
+      totalGamesPlayed
+    ).toFixed(2)}%) <br> Total Games Played: ${
+      userWinRecord + userLoseRecord + userDrawRecord
+    }`;
```
Nice outputting of stats here and good use of the toFixed() method. You have already defined `totalGamesPlayed` in line 77, so you can use it in line 89 as well!

lee-kimmixq commented on project1-sps/script.js:
```diff
@@ -1,4 +1,92 @@
-  return myOutputValue;
+  var myOutputValue;
+  // current game mode
+  if (currentMode === "Enter Name") {
```
Nice game modes!

lee-kimmixq reviewed (commented):
Nice attempt on this project Si Jia! :) I think you have understood the concepts we learnt so far well! Your code is also well-commented which makes it easy to read, and variable naming is fairly consistent following the conventions! I included some suggestions on how to refactor and make your code a little more concise, you can refer to those tips and apply them to your next project! Looking forward to your Beat That!

SJKKSJ committed:
Day 4 & 5 + Project 2 Beat That
- day 4 in class
- day 5 pre class & in class
- Project 2 Base
- Style CSS container color

SJKKSJ committed:
Day 7 Class & Project 2
- day 7 pre and in class
- project beat that index html (comfortable)
- project beat that script edits

SJKKSJ committed:
Project 2 Beat That
- minor changes & checking

SJKKSJ committed:
css final minor trials

SJKKSJ commented:
How many hours did you spend on this assignment?: 12hrs+
What part of the assignment did you spend the most time on?: game modes + game results
What is one error and/or error message you received while working on this assignment, and how did you solve the error? undefined results/output for the player scores, solved by calling the correct array variables
How comfortable did you feel with this assignment? (1-5): 3
Is there anything in this code that you feel pleased about?: that the game works
What's one aspect of your code you would like specific, elaborate feedback on?: game logic conditions

SJKKSJ committed:
Day 7 In Class (Moar Cards)
Moar Cards
- Base
- Display Single Card
- Low Card
- Low Card with Queen Winner

lee-kimmixq commented on project2-beat-that/styles.css:
```diff
@@ -57,7 +57,8 @@ nav ul li a {
 
 .container {
-  background-color: #ffffff;
+  background-image: url("https://www.pngitem.com/pimgs/m/64-646702_dice-png-transparent-big-image-image-dice-transparent.png");
+  /* background-color: #77cdff; */
```
Nice manipulation of CSS :)

lee-kimmixq commented on project2-beat-that/script.js:
```diff
@@ -1,4 +1,124 @@
+  else if (gameMode === "playerOneOrder") {
+    console.log(gameMode);
+    // input validation
+    if (Number(input) !== 1 && Number(input) !== 2) {
+      myOutputValue = `Player One, you rolled ${playerOneNumbers[0]} and ${playerOneNumbers[1]}. <br> Please enter a valid order choice - 1 or 2`;
+      return myOutputValue;
+    }
```
Good input validation here!

lee-kimmixq commented on project2-beat-that/script.js:
```diff
@@ -1,4 +1,124 @@
+      return myOutputValue;
+    }
+    // input order choice of 1 or 2
+    if (Number(input) === 1) {
+      playerOneScore = Number(
+        String(playerOneNumbers[0]) + String(playerOneNumbers[1])
+      );
+      myOutputValue = `Player One, you have chosen your number as ${playerOneScore}. <br> Player Two can roll now.`;
+      // change the game mode to player 2's turn to roll
+      gameMode = "playerTwoRoll";
+    }
+    if (Number(input) === 2) {
+      playerOneScore = Number(
+        String(playerOneNumbers[1]) + String(playerOneNumbers[0])
+      );
+      myOutputValue = `Player One, you have chosen your number as ${playerOneScore}. <br> Player Two can roll now.`;
+      // change the game mode to player 2's turn to roll
+      gameMode = "playerTwoRoll";
+    }
```
- An if-else if block should be better here since they are testing the same variable.
- Good way of concatenating the numbers together here - it shows that you understand the differences in data types :) 
- Note that lines 66-68 and 74-76 are the same, you can take them out of the if statements and just put it below before the game mode ends. That way it'll run in both situations, and you reduce the amount of repeated code you have.

lee-kimmixq commented on project2-beat-that/script.js:
```diff
@@ -1,4 +1,124 @@
+    }
+
+    // player 2 game turn to roll the dice
+  } else if (gameMode === "playerTwoRoll") {
+    console.log(gameMode);
+    var rollDice1 = rollDice();
+    var rollDice2 = rollDice();
+    // to keep record of the dice numbers
+    playerTwoNumbers[0] = rollDice1;
+    playerTwoNumbers[1] = rollDice2;
+    // output dice number rolled and instruction for next game step
+    myOutputValue = `Player Two has rolled ${playerTwoNumbers[0]} and ${playerTwoNumbers[1]}. <br> Please choose order 1 or 2.`;
+    // change to the next game mode
+    gameMode = "playerTwoOrder";
+  }
+
+  // in the next game mode, player 2 to choose the order of their dice numbers
+  else if (gameMode === "playerTwoOrder") {
+    console.log(gameMode);
+    // input validation
+    if (Number(input) !== 1 && Number(input) !== 2) {
+      myOutputValue = `Player Two, you rolled ${playerTwoNumbers[0]} and ${playerTwoNumbers[1]}. <br> Please enter a valid order choice - 1 or 2`;
+    }
+    // player 2 input order choice of 1 or 2
+    if (Number(input) === 1) {
+      playerTwoScore = Number(
+        String(playerTwoNumbers[0]) + String(playerTwoNumbers[1])
+      );
+      myOutputValue = `Player Two, you have chosen your number as ${playerTwoScore}.`;
+      // change the game mode to results
+      gameMode = "gameResults";
+    }
+    if (Number(input) === 2) {
+      playerTwoScore = Number(
+        String(playerTwoNumbers[1]) + String(playerTwoNumbers[0])
+      );
+      myOutputValue = `Player Two, you have chosen your number as ${playerTwoScore}.`;
+      gameMode = "gameResults";
+    }
+  }
```
These parts are similar to `playerOneRoll` and `playerOneOrder` - you can try to refactor into helper functions and call them in the game mode to reduce repetitiveness of code!

lee-kimmixq commented on project2-beat-that/script.js:
```diff
@@ -1,4 +1,124 @@
+      myOutputValue = `Player Two, you rolled ${playerTwoNumbers[0]} and ${playerTwoNumbers[1]}. <br> Please enter a valid order choice - 1 or 2`;
+    }
+    // player 2 input order choice of 1 or 2
+    if (Number(input) === 1) {
+      playerTwoScore = Number(
+        String(playerTwoNumbers[0]) + String(playerTwoNumbers[1])
+      );
+      myOutputValue = `Player Two, you have chosen your number as ${playerTwoScore}.`;
+      // change the game mode to results
+      gameMode = "gameResults";
+    }
+    if (Number(input) === 2) {
+      playerTwoScore = Number(
+        String(playerTwoNumbers[1]) + String(playerTwoNumbers[0])
+      );
+      myOutputValue = `Player Two, you have chosen your number as ${playerTwoScore}.`;
+      gameMode = "gameResults";
+    }
```
Same thing here as lines 62-77, bring the assignment of `myOutputValue` and `gameMode` outside of the if statements so you only need to do them once.

lee-kimmixq commented on project2-beat-that/script.js:
```diff
@@ -1,4 +1,124 @@
+//var totalPlayerOneScore;
+//var totalPlayerTwoScore;
+
+// game results helper function - can it be here??
+var gameResults = function () {
+  var winnerMessage = `Player One's Number: ${playerOneScore} <br> Player Two's Number: ${playerTwoScore}`;
+  var winnerScore = Math.min(playerOneScore, playerTwoScore);
+  console.log(playerOneScore, playerTwoScore, winnerScore);
+  if (playerOneScore == playerTwoScore) {
+    winnerMessage += "<br> It's a draw!";
+  } else if (playerOneScore == winnerScore) {
+    winnerMessage += "<br> Player 2 Wins! 🏆";
+  } else winnerMessage += "<br> Player 1 Wins! 🏆";
+  return winnerMessage;
+};
```
Yes this helper function can be here, and nice job on refactoring the results game mode! Good use of `Math.min` here :)

lee-kimmixq commented on project2-beat-that/script.js:
```diff
@@ -1,4 +1,124 @@
+
+  // to call the game results function to find out the winner
+  else if (gameMode === "gameResults") {
+    var winner = gameResults();
+    return winner;
```
Instead of creating a new variable here and early returning it, you can also just do `myOutputValue = gameResults()`. But good attempt on refactoring here! 💯

lee-kimmixq reviewed (commented):
Awesome job Si Jia! Like before, the comments in your code really helped with the readability. Overall well done, I like how you always try to push yourself beyond base a little, through the use of Math.min, manipulation of CSS, trying to refactor etc.
Just like how you refactored your results game mode, you can challenge yourself to make your main function even smaller (in Project 3) and reduce repeated code :)

SJKKSJ committed:
Project 3 (drawing cards @ start) - Redo/Rearrange
(redo/rearrange)
- Global Variable (game mode and array to record the cards'
- card generation deck and shuffling code from notes
- creating new shuffled deck 
- draw out 2 cards each for player and dealer

SJKKSJ committed:
Project 3 Update
- hand sum display
- hit or stand mode
-

SJKKSJ committed:
Project 3 Update Part 2
- dealer hit or stand mode

SJKKSJ committed:
Project 3 Update 3
- Ace 1 or 11 condition

SJKKSJ committed:
Project 3 Update Revert back to #3
- revert back

SJKKSJ committed:
Project 3 Final Codes (v1)
- update to checkIfBlackjack
- did logic condition for checking if > 21 or not
- FINALISED Codes

SJKKSJ committed:
Project 3 Bug Fix & Cosmetic Changes Part1
- fixing bugs
- gameMode "end"
- cosmetic changes

SJKKSJ committed:
Project 3 Bug Fixes & Cosmetic Changes (#2)
cosmetic changes
bug fixes

SJKKSJ committed:
Project 3 Bug Fixes #3
- bug fixes
- cosmetic updates

SJKKSJ committed:
Project 3 Cosmetic Changes
- Base Code (Final)
- Cosmetic Changes (Some of Comfortable)

SJKKSJ committed:
Re commit (Cosmetic)
re-commit due to error

SJKKSJ committed:
Project 3 CSS Edits
Reset and Edit CSS

SJKKSJ committed:
cosmetic changes
